### DISCUSSION TITLE
How to remove socket.io EIO and other parameters from url

### DISCUSSION THREAD

**Discussion by minikiller** [socketio/socket.io#4101] [Q&A]
When I configure socket-io with the following options: { url: 'ws://localhost:8888', options: {path: '/chatws', transports: ['websocket'], reconnectionAttempts: '3'}}

I get the following error WebSocket connection to 'ws://localhost:8888/chatws/?EIO=3&transport=websocket' failed: Error during WebSocket handshake: Unexpected response code: 501

If I use other frameworks that do not add any parameters then it works.

How does one remove the EIO and all query parameters from the URL?

**Comment by darrachequesne** [comment]
Hi! This is unfortunately not possible, as they are part of how the library works. Could you please explain your use case?

Note: in any case, please remember that the Socket.IO client is not a WebSocket client, it will not be able to reach a plain WebSocket server

Reference: https://socket.io/docs/v4/#what-socketio-is-not
